control-plane-operator/controllers/hostedcontrolplane/kas/config.go

```diff
@@ -173,25 +174,7 @@ func generateConfig(p KubeAPIServerConfigParams, version semver.Version) *kcpv1.
 		args.Set("authentication-token-webhook-config-file", cpath(kasVolumeAuthTokenWebhookConfig().Name, KubeconfigKey))
 		args.Set("authentication-token-webhook-version", "v1")
 	} else {
-		if len(p.Authentication.OIDCProviders) > 0 {
-			provider := p.Authentication.OIDCProviders[0]
-			args.Set("oidc-issuer-url", provider.Issuer.URL)
-			args.Set("oidc-client-id", string(provider.Issuer.Audiences[0]))
-			args.Set("oidc-username-claim", provider.ClaimMappings.Username.Claim)
-			if provider.ClaimMappings.Username.PrefixPolicy == configv1.Prefix &&
-				provider.ClaimMappings.Username.Prefix != nil {
-				args.Set("oidc-username-prefix", provider.ClaimMappings.Username.Prefix.PrefixString)
-			}
-			args.Set("oidc-groups-claim", provider.ClaimMappings.Groups.Claim)
-			args.Set("oidc-groups-prefix", provider.ClaimMappings.Groups.Prefix)
-			for _, cvr := range provider.ClaimValidationRules {
-				// TODO: currently can only support a single required claim because of how kubeAPIServerArgs dedups config fields
-				// In order to specify multiple required claims, the flag must be used multiple times
-				// https://kubernetes.io/docs/reference/access-authn-authz/authentication/#configuring-the-api-server
-				args.Set("oidc-required-claim", fmt.Sprintf("%s=%s", cvr.RequiredClaim.Claim, cvr.RequiredClaim.RequiredValue))
-			}
-			args.Set("oidc-ca-file", oidcCAFile(provider.Issuer.CertificateAuthority.Name))
-		}
+		args.Set("authentication-config", cpath(kasVolumeAuthConfig().Name, AuthenticationConfigKey))
```

Comment:
this should still be conditional. it's reasonable to having an intermediate state with no authentication configured and the kube-apiserver must be able to tolerate that.  SA tokens, client certificates, etc would still function.